### ISSUE TITLE
Support $ref in examples

### ISSUE THREAD

**Issue by vcarluer** [outofcoffee/imposter#49]
I try to reuse examples by placing them in components/examples ans referencing them with $ref but imposter does seams to not support it (example here: https://swagger.io/docs/specification/adding-examples/ 'Reusing Examples')

Do you think iy would be possible to integrate this feature?

**Comment by outofcoffee** [comment]
Yes, this should be relatively quick to do. We can change the resolver mode when the specification is parsed to automatically resolve all `$ref` links. Currently `$ref` is supported, but only in specific scenarios.
